base/number.jl

```diff
@@ -11,6 +11,8 @@ length(x::Number) = 1
 endof(x::Number) = 1
 getindex(x::Number) = x
 getindex(x::Number, i::Integer) = i == 1 ? x : throw(BoundsError())
+getindex(x::Number, i1::Integer, i2::Integer) = 1 == i1 == i2 ? x : throw(BoundsError())
+getindex(x::Number, i1::Integer, i2::Integer, i3::Integer) = 1 == i1 == i2 == i3 ? x : throw(BoundsError())
```

Comment:
Yes, I had deleted them. But now I've been testing them and they are faster than broadcasting (and only a couple of lines of code), I'll put them back.
The changes in number.jl  would still be relevant for broadcasting of user-defined functions.